addons/web/static/src/js/core/utils.js

```diff
@@ -278,6 +278,32 @@ var utils = {
         }
         return "";
     },
+    /**
+     * Gets dataURL (base64 data) from the given file or blob.
+     * Technically wraps FileReader.readAsDataURL in Promise.
+     *
+     * @param {Blob|File} file
+     * @returns {Promise} resolved with the dataURL, or rejected if the file is
+     *  empty or if an error occurs.
+     */
+    getDataURLFromFile: function (file) {
+        return new Promise(function (resolve, reject) {
+            if (!file) {
+                return reject();
+            }
+            var reader = new FileReader();
+            reader.addEventListener('load', function () {
+                resolve(reader.result);
+            });
+            reader.addEventListener('abort', function () {
+                reject();
+            });
+            reader.addEventListener('error', function () {
+                reject();
+            });
```

Comment:
```javascript
reader.addEventListener('error', reject)
```
is shorter and would allow to get the error event on rejections.

Reply:
ok better indeed